src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+        scrollView.layoutParams= scrollViewParams
+        scrollView.setVerticalScrollBarEnabled(false)
+        scrollView.overScrollMode= View.OVER_SCROLL_NEVER
+        scrollView.fullScroll(ScrollView.FOCUS_DOWN)
+
+        scrollView.addView(componentLayout)
+        view.addView(scrollView)
+        addView(view)
+        makeApiCall()
+    }
+
+    //Mark: Function for API Call
+    fun makeApiCall() {
+        val url = "https://api-joy.joyfill.io/v1/documents/"+ SharedData.joyfillDocID
+        val request = Request.Builder()
+            .url(url)
+            .addHeader("Authorization", SharedData.authorizedToken)
+            .addHeader("Content-Type", "application/json")
+            .build()
+
+        client.newCall(request).enqueue(object : okhttp3.Callback {
+            @RequiresApi(Build.VERSION_CODES.P)
+            override fun onResponse(call: okhttp3.Call, response: okhttp3.Response) {
+                val responseBody = response.body?.string()
+                val outputJsonString =  JSONObject(responseBody)
+                val fileArray = outputJsonString.getJSONArray("files")
+                val fields = outputJsonString.getJSONArray("fields")
+
+                for (i in 0 until fileArray.length()){
+                    val pages = fileArray.getJSONObject(i).getJSONArray("pages")
+                    val views = fileArray.getJSONObject(i).getJSONArray("views")
+
+                    if (views.length()>0){
+                        // For Mobile View
+                        for (j in 0 until views.length()){
+                            val view_pages = views.getJSONObject(j).getJSONArray("pages")
+                            for (k in 0 until view_pages.length()){
+                                val fieldPositions =view_pages.getJSONObject(k).getJSONArray("fieldPositions")
+                                for (l in 0 until fieldPositions.length()){
+                                    val component_Type = fieldPositions.getJSONObject(l).get("type")
+                                    val Y_Position =  fieldPositions.getJSONObject(l).getDouble("y")
+                                    yValues_mobile.add(Y_Position.toInt())
+                                    component_type.add(component_Type.toString())
+                                    val componentTypePairedArray = yValues_mobile.zip(component_type)
+                                    val sorted_type = componentTypePairedArray.sortedBy { it.first }
+                                    val sortedType_value = valueForSortedType(sorted_type.toString())
+                                    Sequence_With_Y_Position = sortedType_value
+                                }
+                                formatBlockData(fieldPositions)
+                            }
+                            ShowComponents(Sequence_With_Y_Position, fields)
+                        }
+                    }else {
```

Comment:
This should be ` } else {` with proper spaces and please throughout this code add comments per area where needed. There is so much going on here it can be very confusing to try to and figure out what's doing what.

Reply:
done